libraries/botbuilder/tests/channelServiceRoutes.test.js

```diff
@@ -0,0 +1,776 @@
+            del: sinon.spy(),
+        };
+
+        it('should register webservers', () => {
+            channel.register(server, 'test');
+
+            assert(server.post.calledWith('test/v3/conversations/:conversationId/activities'));
+            assert(server.post.calledWith('test/v3/conversations/:conversationId/activities/:activityId'));
+            assert(server.post.calledWith('test/v3/conversations/:conversationId/activities/history'));
+            assert(server.post.calledWith('test/v3/conversations'));
+            assert(server.post.calledWith('test/v3/conversations/:conversationId/attachments'));
+            assert(server.get.calledWith('test/v3/conversations/:conversationId/activities/:activityId/members'));
+            assert(server.get.calledWith('test/v3/conversations'));
+            assert(server.get.calledWith('test/v3/conversations/:conversationId/members'));
+            assert(server.get.calledWith('test/v3/conversations/:conversationId/pagedmembers'));
+            assert(server.put.calledWith('test/v3/conversations/:conversationId/activities/:activityId'));
+            assert(server.del.calledWith('test/v3/conversations/:conversationId/members/:memberId'));
+            assert(server.del.calledWith('test/v3/conversations/:conversationId/activities/:activityId'));
+        });
+    });
+
+    describe('private functions', () => {
+        const handlers = {};
+        const req = {
+            body: {
+                type: ''
+            },
+            headers: {},
+            params: {},
+            query: {},
+            on(handler, callback) {
+                handlers[handler] = callback;
+            },
+            executeHandler(handler, attr) {
+                if (!!handlers[handler] && typeof handlers[handler] == 'function') {
+                    handlers[handler](attr);
+                }
+            }
+        };
+        const testResource = { id: 'testId' };
+        const errorHandler = ChannelServiceRoutes.handleError;
+        let readActivityStub;
+        let readBodyStub;
+
+        afterEach(() => {
+            ChannelServiceRoutes.handleError = errorHandler;
+            if (!!readActivityStub) {
+                readActivityStub.restore();
+            }
+            if (!!readBodyStub) {
+                readBodyStub.restore();
+            }
+        });
+
+        describe('processSendToConversation()', () => {
+            it('should end successful', (done) => {
+                try {
+                    const res = new MockResponse({
+                        statusCode: 200,
+                        body: testResource
+                    }, done);
+
+                    const service = sinon.createStubInstance(ChannelServiceHandler);
+                    service.handleSendToConversation = sinon.stub().resolves(testResource);
+
+                    const channel = new ChannelServiceRoutes(service);
+                    channel.processSendToConversation(req, res);
```

Comment:
Is an assert statement missing here?

Reply:
when processSendToConversation is called it get the **res** passed, which is an instance of MockResponse, declared in line 5, it will assert when the end() method is called, which is the last method the processSendToConversation will call. We did this because the methods can't be awaited and can't be handled as promises so if we assert this outside the asserts run before the end() method was called and it asserts as false.

Reply:
Ok, I see. Thanks!